### DISCUSSION TITLE
Do we know how many CLOPS?

### DISCUSSION THREAD

**Discussion by InfamousPlatypus** [Spooky-Manufacturing/8Q#25] [Q&A] [answered]
How many gates can this design execute per second? I feel like having some benchmarks ready could be useful and am working on getting documents together for that.
Obviously, many will depend on the skill of the individual, but I believe CLOPS is one where we can at least get accurate guesses for the maximum.
Thanks!

**Comment by NoahGWood** [comment, marked as answer]
The gate speed is 150ns, the actual timing of the mems is ~75ns, but we add a little bit extra delay to help with timing. So our physical limit would be 6 million operations per second before we have to start scaling to chip size.

The slowest ops on this processor is the measurement and the single-photon source with a 4us and 10us cooldown times respectively. That's 14 microseconds which is the minimum time you need in between each program. It looks like there's 16 gate operations which means the algorithm would be:

14us + 150n(16) = 16.4us
1/16.4us * 1000000/1 = 60975.61= ~60,000kHz

So, ~60,000 clops? I don't like this unit of measure though as it's not really that informative and it seems kinda arbitrary. Realistically measured clops will be less because we have probabilistic gates which approach n=1 but it's not guaranteed that every operation will be successful, and as you add multiple operations that rate of success decreases which requires you to repeat operations which slows you down

**Reply by InfamousPlatypus** [reply]
No, it's not the greatest measurement, but I understand why some people would want it. (Me, I'm some. I think, I have an equation somewhere I need this for.)
Thanks!
